Content.Server/Doors/Systems/FirelockSystem.cs

```diff
@@ -51,36 +43,13 @@ public override void Update(float frameTime)
 
             _accumulatedTicks = 0;
 
-            var airtightQuery = GetEntityQuery<AirtightComponent>();
-            var appearanceQuery = GetEntityQuery<AppearanceComponent>();
-            var xformQuery = GetEntityQuery<TransformComponent>();
-            var pointLightQuery = GetEntityQuery<PointLightComponent>();
-
+            // Go through all of the firelocks and re-close them if they've been pried open.
             var query = EntityQueryEnumerator<FirelockComponent, DoorComponent>();
             while (query.MoveNext(out var uid, out var firelock, out var door))
```

Comment:
Why is this query still here? Shouldn't it just close the door when IsLocked gets set or something else happens, instead of uneccesarily iterating over all firelocks every tick? Also if it is needed, it should defer getting the door component.

Reply:
This is here because anyone can pry the airlock at any time. The AfterPry sets a timeout after which the door should automatically close, but since I'm not aware that we have timer events this needs to check that the timer has elapsed on a "IsLocked" airlock.